girder/girder_large_image/web_client/vue/components/PresetsMenu.vue

```diff
@@ -0,0 +1,148 @@
+            }).then((presets) => {
+                if (presets) {
+                    this.availablePresets = presets
+                }
+            })
+        },
+        addPreset() {
+            const newPreset = {
+                'name': this.newPresetName || this.generatedPresetName,
+                'mode': this.currentMode,
+                'frame': this.currentFrame,
+                'style': this.currentStyle,
+            }
+            if (this.availablePresets.find((p) => p.name === newPreset.name)) {
+                this.errorMessage = `There is already a preset named "${newPreset.name}".`
+            } else {
+                this.availablePresets.push(newPreset)
+                this.selectedPreset = newPreset.name
+                this.savePresetsList()
+            }
+            this.newPresetName = undefined
+        },
+        deleteSelectedPreset() {
+            this.availablePresets = this.availablePresets.filter((p) => p.name !== this.selectedPreset)
+            this.selectedPreset = undefined
+            this.savePresetsList()
+        },
+        savePresetsList() {
+            restRequest({
+                type: 'PUT',
+                url: 'item/' + this.itemId + '/internal_metadata/presets',
+                data: JSON.stringify(this.availablePresets),
+                contentType: 'application/json',
+            })
+        }
+    },
+    computed: {
+        generatedPresetName() {
+            let name = `${this.currentMode.name} control - Frame ${this.currentFrame}`;
+            if (this.currentMode.id === 2) {
+                name = `${this.currentStyle.bands.length} channels`
+            } else if (this.currentMode.id === 3) {
+                name = `${this.currentStyle.bands.length} bands`
+            }
+            return name;
+        }
+    },
+    watch: {
+        newPresetName() {
+            if (this.newPresetName){
+                this.errorMessage = undefined;
+            }
+        },
+        selectedPreset() {
+            if (this.selectedPreset) {
+                const preset = this.availablePresets.find((p) => p.name === this.selectedPreset)
+                if (preset.mode && preset.mode.id !== undefined) {
+                    this.$emit('setCurrentMode', preset.mode);
+                }
+                if (preset.frame !== undefined) {
+                    this.$emit('setCurrentFrame', preset.frame)
+                }
+                if (preset.style && Object.keys(preset.style.bands).length) {
+                    this.$emit('updateStyle', preset.mode.id, preset.style)
+                }
+            }
+        }
+    },
+    mounted() {
+        this.getPresets()
+    },
+}
+</script>
+
+<template>
+    <div class="presets-menu">
+        Image View Presets:
+        <select v-model="selectedPreset">
+            <option v-if="availablePresets.length === 0" :value="undefined" selected disabled>
+                No presets available.
```

Comment:
For appearance consistency, I think we should remove the period at the end of `No presets available.`

Reply:
Ok, changed in [eee3513](https://github.com/girder/large_image/pull/1237/commits/eee351327fa1cff7c07b01107958740a95bd9cc9)